beacon-chain/state/stategen/replay.go

```diff
@@ -235,7 +235,7 @@ func (s *State) lastSavedBlock(ctx context.Context, slot types.Slot) ([32]byte,
 		return gRoot, 0, nil
 	}
 
-	lastSaved, err := s.beaconDB.HighestSlotBlocksBelow(ctx, slot+1)
+	lastSaved, err := s.beaconDB.HighestSlotBlocksBelow(ctx, slot)
```

Comment:
Tracing back to `State.StateBySlot` I see there's quite a bit of code using the current method. Is there any way we could  be more confident that none of these callers are relying on the previous incorrect behavior?

Reply:
I checked all its usages, they are all debugging related